builtin/providers/aws/resource_aws_elasticache_cluster.go

```diff
@@ -336,6 +337,13 @@ func CacheClusterStateRefreshFunc(conn *elasticache.ElastiCache, clusterID, give
 
 		// return given state if it's not in pending
 		if givenState != "" {
+			// loop the nodes and check their status as well
+			for _, n := range c.CacheNodes {
```

Comment:
This range will not cover the case where `len(c.CacheNodes) < len(c.NumCacheNodes)`, which is I believe the problematic scenario, yeah?